### DISCUSSION TITLE
Alias to type lambda cannot be used without application

### DISCUSSION THREAD

**Discussion by s5bug** [scala/scala3#17669] [Feature Requests]
## Compiler version

3.2.0, 3.2.1-RC2, 3.2.1

## Minimized code

```Scala
type →[-L <: AnyKind, +R <: AnyKind] = [_ <: L] =>> R

type F = Any → Any
```

## Output

```scala
-- Error: ----------------------------------------------------------------------
1 |type F = Any → Any
  |         ^^^^^^^^^
  |         type → takes type parameters
```

## Expectation

Scala allows this usage, especially in the context of
```scala
trait Free[F <: ((Any → Any) → Any), A]
```
where
```scala
trait Free[F <: /* giant mess of equivalent type lambda */, A]
```
is equivalent and does compile.

**Comment by smarter** [comment]
I'm moving this to feature-request because this was an intentional change, although it could change back again as discussed in https://github.com/lampepfl/dotty/pull/9090

**Comment by s5bug** [comment]
As a question, why does the current error appear at usage and not at declaration?

**Comment by smarter** [comment]
You mean, why is there no error in the definition of `type →` ? Because it defines a type lambda which is allowed, you can also write:
```scala
type F = [X] =>> (Any → Any)[X]
```
But the eta-expansion is not performed automatically by the compiler
